context/BooksContext.ts

```diff
@@ -0,0 +1,3 @@
+import React, { createContext } from "react"
```

Comment:
You can actually have multiple `useState` in a context.

I have a point of reference here for doing `ContextAPI` I'll share it to you.

```
import React, { useRef, useState } from "react"

import { ConfirmationModal } from "components/modals"

type TextAlignments = "center" | "left" | "right"

interface ConfirmationModalProperties {
  title: string
  cancelText?: string
  confirmText?: string
  body: React.ReactNode
  textAlignment: TextAlignments
}

const ConfirmationModalContext = React.createContext<ContextType | undefined>(
  undefined
)

const useConfirmationModals = () => {
  const context = React.useContext(ConfirmationModalContext)
  if (!context) {
    throw new Error(
      `useConfirmationModals must be used within an ConfirmationModalContext`
    )
  }
  return context
}

interface ContextType {
  showConfirmation: (props: ConfirmationModalProperties) => Promise<unknown>
}

const ConfirmationModalContextProvider = (props: any) => {
  const [showConfirmationModal, setShowConfirmationModal] = useState(false)
  const [title, setTitle] = useState("")
  const [body, setBody] = useState<React.ReactNode>()
  const [cancelText, setCancelText] = useState<string>()
  const [confirmText, setConfirmText] = useState<string>()
  const [textAlignment, setTextAlignment] = useState<TextAlignments>()
  const resolver = useRef<any>()

  const handleShow = (values: ConfirmationModalProperties) => {
    const { title, body, cancelText, confirmText, textAlignment } = values
    setTitle(title)
    setBody(body)
    setCancelText(cancelText)
    setConfirmText(confirmText)
    setShowConfirmationModal(true)
    setTextAlignment(textAlignment)

    return new Promise(function (resolve) {
      resolver.current = resolve
    })
  }

  const handleOk = () => {
    resolver.current && resolver.current(true)
    setShowConfirmationModal(false)
  }

  const handleCancel = () => {
    resolver.current && resolver.current(false)
    setShowConfirmationModal(false)
  }

  const value: ContextType = {
    showConfirmation: handleShow,
  }
  return (
    <ConfirmationModalContext.Provider value={value}>
      {props.children}
      <ConfirmationModal
        open={showConfirmationModal}
        onConfirm={handleOk}
        onClose={handleCancel}
        confirmText={confirmText}
        cancelText={cancelText}
        textAlignment={textAlignment}
        title={title}
      >
        {body}
      </ConfirmationModal>
    </ConfirmationModalContext.Provider>
  )
}

export {
  ConfirmationModalContext,
  ConfirmationModalContextProvider,
  useConfirmationModals,
}
```